parser/agent.go

```diff
@@ -363,12 +380,34 @@ func InstrumentMain(mainFunctionNode dst.Node, manager *InstrumentationManager,
 // with a newrelic transaction. All transactions are assumed to be named "txn"
 func NoticeError(manager *InstrumentationManager, stmt dst.Stmt, c *dstutil.Cursor, tracing *tracingState) bool {
 	switch nodeVal := stmt.(type) {
+	case *dst.IfStmt:
+		if binExpr, ok := nodeVal.Cond.(*dst.BinaryExpr); ok {
+			if _, ok := binExpr.X.(*dst.Ident); ok {
+				errExpr := dst.Expr(nil)
```

Comment:
Lets check if one of the parameters of the if statement is of type errors.Error. There is a tool in my PR that will let you easily do that.